domains/auth/Auth.API/API/Controllers/LoginController.cs

```diff
@@ -28,4 +51,92 @@ public LoginController(IOidcService oidcService)
 
         return oidcService.CreateAuthorizationUri(state);
     }
+
+    [HttpGet]
+    [Route("/oidc/login/callback")]
+    public async Task<ActionResult<NextStep>> CallbackAsync(
+        OidcCallbackParams oidcCallbackParams,
+        [FromServices] ICryptographyFactory stateCryptography,
+        [FromServices] ICryptographyFactory IdtokenCryptography
```

Comment:
I believe only one factory is needed here

Reply:
Both are being used. We need to encrypt the IdToken and the state.